File: jscomp/stdlib/callback.ml
(***********************************************************************)
(*                                                                     *)
(*                                OCaml                                *)
(*                                                                     *)
(*            Xavier Leroy, projet Cristal, INRIA Rocquencourt         *)
(*                                                                     *)
(*  Copyright 1996 Institut National de Recherche en Informatique et   *)
(*  en Automatique.  All rights reserved.  This file is distributed    *)
(*  under the terms of the GNU Library General Public License, with    *)
(*  the special exception on linking described in file ../LICENSE.     *)
(*                                                                     *)
(***********************************************************************)

(* Registering OCaml values with the C runtime for later callbacks *)

external register_named_value : string -> Obj.t -> unit
                              = "caml_register_named_value"

let register name v =
  register_named_value name (Obj.repr v)

let register_exception name (exn : exn) =
  let exn = Obj.repr exn in
  let slot = if Obj.tag exn = Obj.object_tag then exn else Obj.field exn 0 in
  register_named_value name slot

File: lib/es6/callback.js


import * as Obj from "./obj.js";

function register(name, v) {
  return /* () */0;
}

function register_exception(name, exn) {
  (exn.tag | 0) === Obj.object_tag ? exn : exn[0];
  return /* () */0;
}

export {
  register ,
  register_exception ,
  
}
/* No side effect */
